java-frontend/src/main/java/org/sonar/java/regex/RegexParser.java

```diff
@@ -81,6 +83,7 @@ public RegexParser(RegexSource source, FlagSet initialFlags) {
   }
 
   public RegexParseResult parse() {
+    this.activeFlags = initialFlags;
```

Comment:
It probably makes sense to initialize `activeFlags` in the constructor rather than `parse` and to get rid of `initialFlags` altogether (this would also entail getting rid of `RegexParseResult.initialFlags`, which should be fine because we can just use `parseResult.getResult().activeFlags()` instead).

Reply:
I removed the field `initialFlags` but keept the getter `getInitialFlags()`.